.pipelines/pr-e2e.yaml

```diff
@@ -0,0 +1,120 @@
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    exclude:
+    - docs/*
+
+pr:
+  branches:
+    include:
+    - master
+  paths:
+    exclude:
+    - docs/*
+
+resources:
+  containers:
+  - container: dev1
+    image: quay.io/deis/go-dev:v1.18.2
+
+jobs:
+- job: unit_tests
+  timeoutInMinutes: 60
+  cancelTimeoutInMinutes: 5
+  strategy:
+    maxParallel: 0
+  pool:
+      vmImage: 'Ubuntu 16.04'
+
+  container: dev1
+
+  variables:
+    GOBIN:  '$(GOPATH)/bin' # Go binaries path
+    GOROOT: '/usr/local/go' # Go installation path
+    GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
+    modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code
+
+  steps:
+  - script: |
+      mkdir -p '$(GOBIN)'
+      mkdir -p '$(GOPATH)/pkg'
+      mkdir -p '$(modulePath)'
+      shopt -s extglob
+      mv !(gopath) '$(modulePath)'
+      echo '##vso[task.prependpath]$(GOBIN)'
+      echo '##vso[task.prependpath]$(GOROOT)/bin'
+    displayName: 'Set up the Go workspace'
+  - script: make bootstrap
+    displayName: Install dependencies
+    workingDirectory: $(modulePath)
+  - script: make test-style
+    displayName: Run linting rules
+    workingDirectory: $(modulePath)
+  - script: make validate-dependencies
+    displayName: Check if imports, Gopkg.toml, and Gopkg.lock are in sync
+    workingDirectory: $(modulePath)
+  - script: make build-cross
+    displayName: Build cross-architectural binaries
+    workingDirectory: $(modulePath)
+  - script: make coverage
+    displayName: Run unit tests and calculate coverage
+    workingDirectory: $(modulePath)
+
+- template: e2e-job-template.yaml
+  parameters:
+    name: 'default_linux_e2e'
+    apimodel: 'examples/kubernetes.json'
+
+- template: e2e-job-template.yaml
+  parameters:
+    name: 'k8s_1dot10_release_e2e'
```

Comment:
nit: can we have an underscore instead of `dot`? More readable that way IMO

Reply:
If we don't like "dot" I vote for a period (`.`)

Reply:
period isn't a valid char in the name (that was my first choice too)

I can change it to underscore